### PR TITLE
PAB-4074: Change alarm text for infinite loops

### PR DESCRIPTION
**Pull request by kailol-sag**
Need to make sure that the alarm text in the doc is the same as in the dev sources. See https://github.softwareag.com/IOTA/apama-in-c8y/pull/2295

### CODE REVIEW

File: content/streaming-analytics/troubleshooting-bundle/alarms.md [hunk 1]
```diff
@@ -248,7 +248,7 @@ monitor Sample{
 Apama-ctrl generates the following alarm for the above example:
 
 - Alarm type: `APAMA_CTRL_WARN_<HASHCODE>`
-- Alarm text: &lt;EPLAppName&gt;.&lt;monitorName&gt; - context '&lt;contextName&gt;' has been processing a single event for a long time.
+- Alarm text: &lt;EPLAppName&gt;.&lt;monitorName&gt; - This EPL app is running in an infinite loop and impedes the operation of your other apps. It is recommended that you deactivate and diagnose this app.
 - Alarm severity: WARNING
 
 You can diagnose the issue by the monitor name and context name given in the alarm.
```
Comment (kailol-sag) [inline content/streaming-analytics/troubleshooting-bundle/alarms.md:254]: See also my comment/question on https://itrac.eur.ad.sag/browse/PAB-4074 :

The doc still says "You can diagnose the issue by the monitor name and context name given in the alarm.", but you removed "context '<contextName>' " from the alarm. So which other doc changes need to be done?

File: content/streaming-analytics/troubleshooting-bundle/alarms.md [hunk 2]
```diff
@@ -225,7 +225,7 @@ You can diagnose the issue by the monitor name and line number given in the alar
 For more details, you can also check the Apama logs if the tenant has the "microservice hosting" feature enabled. Alarms of this type should be fixed as a priority as these uncaught exceptions will terminate the execution of that monitor instance, which will typically mean that your app is not going to function correctly. This might even lead to a correlator crash if not handled properly.
 
 <a name="apama_ctrl_warn"></a>
-#### An EPL file blocks the correlator context for too long
+#### An EPL app is running in an infinite loop
```
Comment (BeateRixen) [inline content/streaming-analytics/troubleshooting-bundle/alarms.md:228]:
```suggestion
#### An EPL app runs in an infinite loop
```

File: content/streaming-analytics/troubleshooting-bundle/alarms.md [hunk 3]
```diff
@@ -248,7 +248,7 @@ monitor Sample{
 Apama-ctrl generates the following alarm for the above example:
 
 - Alarm type: `APAMA_CTRL_WARN_<HASHCODE>`
-- Alarm text: &lt;EPLAppName&gt;.&lt;monitorName&gt; - context '&lt;contextName&gt;' has been processing a single event for a long time.
+- Alarm text: &lt;EPLAppName&gt;.&lt;monitorName&gt; - This EPL app is running in an infinite loop and impedes the operation of your other apps. It is recommended that you deactivate and diagnose this app.
```
Comment (BeateRixen) [inline content/streaming-analytics/troubleshooting-bundle/alarms.md:251]:
```suggestion
- Alarm text: &lt;EPLAppName&gt;.&lt;monitorName&gt; - This EPL app runs in an infinite loop and impedes the operation of your other apps. It is recommended that you deactivate and diagnose this app.
```

File: content/streaming-analytics/troubleshooting-bundle/alarms.md (hunk 1 above)
Comment (rthrippleton) [reply content/streaming-analytics/troubleshooting-bundle/alarms.md:254]: Fair point. Thinking out loud - we've removed a bit of info from the alarm text (the context name), which seems bad at first glance. However:
1. This means that a badly behaved multi-context application doesn't create alarm spam (lots of _different_ alarms with nearly-identical text)
2. To diagnose the application, it's almost certain that you'll need the microservice log anyway, which will tell you the context.
So I think I'm happy with this. @kailol-sag - I think we need text suggesting that you'll need to look at the microservice log, with whatever xrefs are appropriate.

File: content/streaming-analytics/troubleshooting-bundle/alarms.md (hunk 2 above)
Comment (rthrippleton) [reply content/streaming-analytics/troubleshooting-bundle/alarms.md:228]: Had this discussion elsewhere, but that doesn't quite scan right - as a native speaker, it feels wrong, but I think _formally_ this needs to be in the continuous present tense :-D. So I think "is running" is correct here.

File: content/streaming-analytics/troubleshooting-bundle/alarms.md (hunk 3 above)
Comment (rthrippleton) [inline content/streaming-analytics/troubleshooting-bundle/alarms.md:251]: Thought some more about it, and realised that in practice this alarm isn't a 100% guarantee of an _infinite_ loop. I'd like to hedge slightly e.g. "This EPL app is probably in an infinite or long-running loop".

Comment (kailol-sag) [reply content/streaming-analytics/troubleshooting-bundle/alarms.md:251]: @BeateRixen , it's funny, I suggested exactly the same when we were discussing the alarm text in https://itrac.eur.ad.sag/browse/PAB-3509. But @rthrippleton felt that "runs" is not right and asked for "running".
Also, this is an alarm text that is defined in the dev sources and we must not use a different text here. Therefore, I won't commit your suggestion.

File: content/streaming-analytics/troubleshooting-bundle/alarms.md (hunk 1 above)
Comment (kailol-sag) [reply content/streaming-analytics/troubleshooting-bundle/alarms.md]: I removed mention of the context name. 
Added a link to the log files section at the end of the next paragraph which talks about "Apama logs". Hmmm, is "Apama logs" the right term here or should it be "log files of the Apama-ctrl microservice" ???

File: content/streaming-analytics/troubleshooting-bundle/alarms.md (hunk 3 above)
Comment (kailol-sag) [reply content/streaming-analytics/troubleshooting-bundle/alarms.md]: I changed the alarm text as suggested. But then you also need to update it on https://github.softwareag.com/IOTA/apama-in-c8y/pull/2295

Comment (kailol-sag) [reply content/streaming-analytics/troubleshooting-bundle/alarms.md]: @rthrippleton , do you also want to change the heading and first sentence of this section to include the "long-running"?

File: content/streaming-analytics/troubleshooting-bundle/alarms.md (hunk 1 above)
Comment (kailol-sag) [reply content/streaming-analytics/troubleshooting-bundle/alarms.md]: Hi @rthrippleton , I'm still waiting for an answer to the above question: is "Apama logs" the right term?

Comment (rthrippleton) [reply content/streaming-analytics/troubleshooting-bundle/alarms.md]: I think "log files of the Apama-ctrl microservice" has to be it - `Apama` is a product name we're not supposed to use, but `Apama-ctrl` is a technical microservice name :-)

Comment (kailol-sag) [reply content/streaming-analytics/troubleshooting-bundle/alarms.md]: Thanks @rthrippleton , I've changed that in 2 places, and also added the link to the other place.

**Review by rthrippleton — APPROVED** [review]

**Review by BeateRixen — APPROVED** [review]

**Pull request merged**